test/byContract/IexecPoco/IexecPoco1.test.ts

```diff
@@ -408,38 +504,312 @@ describe('IexecPoco1', () => {
-            // No request order
-            ['requester', 'app', 'workerpool', 'dataset'].forEach((assetName) => {
-                // Filter irrelevant cases. E.g. no need to change the app address in the app order.
+        // No restrictions in request order.
+        ['app', 'dataset', 'workerpool'].forEach((orderName) => {
+            ['app', 'dataset', 'workerpool', 'requester'].forEach((assetName) => {
+                // Filter irrelevant cases (e.g. app - app).
                 if (orderName.includes(assetName)) {
                     return;
                 }
-                it(`[TEE] Should fail when ${orderName} order mismatch ${assetName} restriction (EOA, SC)`, async function () {
+                it(`Should fail when ${orderName} order mismatch ${assetName} restriction (EOA, SC)`, async function () {
                     const message = revertMessages[orderName][assetName];
                     // EOA
-                    orders[orderName][assetName + 'restrict'] = randomAddress; // e.g. orders['app']['apprestrict'] = 0xEOA
+                    orders[orderName][assetName + 'restrict'] = randomAddress; // e.g. orders.app.datasetrestrict = 0xEOA
                     await expect(iexecPoco.matchOrders(...orders.toArray())).to.be.revertedWith(
                         message,
                     );
                     // SC
-                    orders[orderName][assetName + 'restrict'] = randomContract.address; // e.g. orders['app']['apprestrict'] = 0xSC
+                    orders[orderName][assetName + 'restrict'] = randomContract.address; // e.g. orders.app.datasetrestrict = 0xSC
                     await expect(iexecPoco.matchOrders(...orders.toArray())).to.be.revertedWith(
                         message,
                     );
                 });
             });
         });
 
-        it('[TODO] Should match orders with replication', () => {});
+        it('Should fail when app is not registered', async function () {
+            orders.app.app = randomContract.address; // Must be an Ownable contract.
+            orders.requester.app = randomContract.address;
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray())).to.be.revertedWith(
+                'iExecV5-matchOrders-0x20',
+            );
+        });
+
+        it('Should fail when invalid app order signature from EOA', async function () {
+            await signOrders(iexecWrapper.getDomain(), orders, ordersActors);
+            orders.app.sign = randomSignature; // Override signature.
+            await expect(iexecPocoAsRequester.matchOrders(...orders.toArray())).to.be.revertedWith(
+                'iExecV5-matchOrders-0x21',
+            );
+        });
+
+        it('Should fail when invalid app order signature from SC', async function () {
+            await signOrders(iexecWrapper.getDomain(), orders, ordersActors);
+            orders.app.sign = randomSignature; // Override signature.
```

Comment:
I wonder if it is less error prone that way?
```
// app order not signed
await signOrder(domain, datasetOrder, datasetProvider);
await signOrder(domain, workerpoolOrder, scheduler);
await signOrder(domain, requestOrder, requester);
```

Reply:
I personally find it less noisy with only one line to sign all orders and one line to highlight which signature is invalid.

Reply:
Ok no worry ;)